### DISCUSSION TITLE
Building vs Building Part

### DISCUSSION THREAD

**Discussion by Azbesciak** [OvertureMaps/data#288] [Discussion]
Hi, I am not sure reading the doc, also I did not find any cases for that, so please clarify.
We have `building` and `building_part` features. `building_part` has also a field `building_id`. Does it mean we have one to many relation between `building` and `building_part`, and in `building` we will have the entity for the given id? So far I rather see that as some virtual entity, which only exists after collecting all parts - that given id does not really has any existing id as a `building` entity. But I may be wrong - I would even assume from the description, I am.

Thank you!

**Comment by jwass** [comment]
Hi @Azbesciak - You are correct that the `building_part.building_id` field is a foreign key reference to the `building` entity. There's a backward relation in `building.has_parts` which indicates whether any `building_part`s exist that point back to this building.

The `building` entities are the defining record of information about the building - containing things like the type of building (in `class` and `subclass`) and the building's name. The `building_part`s are the 3d representation. But if you only want the 2d geometry, you can find it in `building.geometry`.

**Reply by Azbesciak** [reply]
Ok, thank you - I had my assumptions mostly investigating https://explore.overturemaps.org/#17.75/52.404441/16.952647, where if I toggle buildings or building parts, these seem to be exclusive :)

Thank you once again!
